src/main/java/frc/robot/commands/StorageRunCommand.java

```diff
@@ -14,17 +15,26 @@ public StorageRunCommand() {
 
     @Override
     public void execute() {
-        storage.run();
+        storage.runStorage();
+        storage.runTransition();
     }
 
     @Override
     public boolean isFinished() {
-        return false;
+        if (storage.noteExitingStorage()) {
+            seenNote = true;
+            return false;
+        } else if (!storage.noteEnteringStorage() && seenNote) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
I would relook this logic because it will always return false. You are changing the "seenNote" variable if and only if the note is exiting the storage, so there is no way for the second condition to execute.